lib/client/workloadendpoint.go

```diff
@@ -63,7 +64,16 @@ func (w *workloadEndpoints) Apply(a *api.WorkloadEndpoint) (*api.WorkloadEndpoin
 
 // Delete deletes an existing workload endpoint.
 func (w *workloadEndpoints) Delete(metadata api.WorkloadEndpointMetadata) error {
-	return w.c.delete(metadata, w)
+	// Validate the supplied Metadata.
+	if err := validator.Validate(metadata); err != nil {
+		return err
+	}
+
+	k, err := w.convertMetadataToKey(metadata)
+	if err != nil {
+		return err
+	}
+	return w.c.Backend.Delete(&model.KVPair{Key: k, Revision: metadata.Revision})
```

Comment:
mind sticking a comment so others don't trip when they modify `delete` in client.go and dont see it being reflected for WEP?